ludwig/trainers/trainer.py

```diff
@@ -765,19 +801,28 @@ def train(self, training_set, validation_set=None, test_set=None, save_path="mod
             if test_summary_writer is not None:
                 test_summary_writer.close()
 
-            if self.is_coordinator() and not self.skip_save_progress:
+            if not self.skip_save_progress:
                 checkpoint_manager.close()
 
         # Load the best weights from saved checkpoint
-        if self.is_coordinator() and not self.skip_save_model:
-            self.model.load(save_path)
+        state_dict = None
+        if not self.skip_save_model:
+            state_dict = checkpoint_manager.get_best_checkpoint_state_for_inference(self.return_device)
+            if not return_state_dict:
+                if self.distributed.is_model_parallel():
+                    # Assume the full weights cannot fit in memory on GPU
+                    self.model = self.model.cpu()
+                self.model.load_state_dict(state_dict)
+        elif return_state_dict:
+            state_dict = self.model.cpu().state_dict()
+        return_value = self.model if not return_state_dict else state_dict
```

Comment:
Are you able to leave a comment on why/when we'd want to return the model state dict as opposed to the model with the loaded state dict?

Reply:
I assume it is because each of the workers can return their individual state dicts (because of the model parallelism), and then it's all combined back on the coordinator, but I can't seem to see the conditional here for is_coordinator() to load the combined state dicts back into the model object

Reply:
Ray sends the state dict back to the caller, not the model, so loading the weights back into the model is wasteful.